pgjdbc/src/main/java/org/postgresql/jdbc/PgCallableStatement.java

```diff
@@ -691,7 +696,120 @@ public Reader getCharacterStream(String parameterName) throws SQLException {
   }
 
   public <T> T getObject(int parameterIndex, Class<T> type) throws SQLException {
-    if (type == ResultSet.class) {
+    if (type == null) {
+      throw new SQLException("type is null");
+    }
+    checkIndex(parameterIndex);
+    int sqlType = functionReturnType[parameterIndex - 1];
+    if (type == BigDecimal.class) {
+      if (sqlType == Types.NUMERIC || sqlType == Types.DECIMAL) {
+        return type.cast(getBigDecimal(parameterIndex));
+      } else {
+        throw new SQLException("conversion to " + type + " from " + sqlType + " not supported");
+      }
+    } else if (type == String.class) {
+      if (sqlType == Types.CHAR || sqlType == Types.VARCHAR) {
+        return type.cast(getString(parameterIndex));
+      } else {
+        throw new SQLException("conversion to " + type + " from " + sqlType + " not supported");
+      }
+    } else if (type == Boolean.class) {
+      if (sqlType == Types.BOOLEAN || sqlType == Types.BIT) {
```

Comment:
Yes, the JDBC spec is annoyingly  ambiguous with the whole BIT thing, from one side the spec suggest mapping `BIT` to `boolean`, which clearly  is just for `BIT(1)` but for the SQL:92 spec the BIT data type is a completely different beast, in fact it's called bit string, for which the JDBC spec says nothing.

Since right now the BIT(>1) it's broken, I guess the proper fix should be in another PR, for the record Npgsql threats bit(1) as boolean and bit(>1) as BitString (in .NET), and MySQL JDBC threat bit(>1) as byte[].